### DISCUSSION TITLE
Faster-Whisper-XXL r192.3.4 won't use GPU, but r192.3 works fine

### DISCUSSION THREAD

**Discussion by limheat** [Purfview/whisper-standalone-win#249] [Q&A] [answered]
Faster-Whisper-XXL r192.3.4 won't use GPU. Already tried `--device cuda`, didn't work.
Faster-Whisper r192.3 works completely fine on GPU.
Btw, is it worth switching to 192.3.4? I prioritize transcription quality over speed.
(Windows)

**Comment by Purfview** [comment]
> won't use GPU. Already tried --device cuda, didn't work.

Post screenshot of error you get, and post the model of your GPU.

**Reply by limheat** [reply]
I was not able to run faster-whisper-xxl.exe normally. Somehow it always tries to download the Medium model for me, even though I have large-v2 in the correct directory, and it definitely worked on large-v2 last time I tried.

Last time (3-4 weeks ago) I ran faster-whisper-xxl.exe in cmd it just said something something "running on: CPU" in the cmd windows, and proceeded to transcribe the file normally. When using Faster-Whisper r192.3, it says "`Standalone Faster-Whisper r192.3 running on: CUDA`". I hope that's enough information.

My gpu is GTX 1660 (not listed on the cuda capability table on nvidia's website) GeForce GTX 980 Ti supports cuda 5.2, its computing calability is the closest to my gpu on the list as far as I know.

**Reply by Purfview** [reply]
Post screenshot of the error when you use `--device cuda`.
What other files do you have besides faster-whisper-xxl.exe in same folder?

> Somehow it always tries to download the Medium model for me.

Post the screenshots of that.

**Reply by limheat** [reply]
I upgraded the gpu driver and it works now, on cuda.
The prompt was `faster-whisper-xxl.exe E:\test1.mp4 -o E: --model large-v2 --language en --print_progress --standard`

The reason it tried to download the medium model for me was bc I initially used `faster-whisper-xxl.exe "E:\test1.mp4" -o "E:\"` in the command and according to the error message in cmd, which said 

> Traceback (most recent call last):
>   File "D:\whisper-fast\_XXL\__main__.py", line 1633, in <module>
>   File "D:\whisper-fast\_XXL\__main__.py", line 1598, in cli
>   File "os.py", line 223, in makedirs
> OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: 'E:\\" --model medium --language en --print_progress --standard'
> [6424] Failed to execute script '__main__' due to unhandled exception!

apparently everything after  `"E:\` was ignored. Then the program used model medium by default.

Seems like I shouldn't have used quotation marks around the directory. I would never have thought of that, it always used to work when using quotation marks on longer directories like `"E:\podcasts"`.

**Reply by 1001ruchka** [reply]
> Seems like I shouldn't have used quotation marks around the directory. I would never have thought of that, it always used to work when using quotation marks on longer directories like `"E:\podcasts"`.

Yes, it is.
Make it a rule to enclose the file name and path in quotation marks. 
And I would like to point out that your command line does not use any of the faster-whisper-xxl features.  With this command line you can use the standard version of faster-whisper.
If you don't use --vad_alt_method and/or --ff_mdx_kim2 - it doesn't make sense to use XXL version.

**Reply by limheat** [reply]
OK, Got it👍

**Reply by Purfview** [reply]
Path can't end with `\` backslash or you'll get syntax error.

**Reply by limheat** [reply]
That explains a lot. Sadly I was unaware of that and had to learn the hard way...

**Reply by Purfview** [reply]
> Sadly I was unaware of that and had to learn the hard way...

You could learn it soft way if you could articulate your issues two weeks ago.

**Comment by iTula** [comment]
The same situation for me. And the error is this:
Traceback (most recent call last):
  File "D:\whisper-fast\_XXL\__main__.py", line 1668, in <module>
  File "D:\whisper-fast\_XXL\__main__.py", line 1505, in cli
  File "faster_whisper\transcribe.py", line 143, in __init__
RuntimeError: CUDA failed with error CUDA driver version is insufficient for CUDA runtime version
[42128] Failed to execute script '__main__' due to unhandled exception!

**Reply by iTula** [reply]
I just updated the Nvidia cuda software and it works with cuda.

**Reply by Purfview** [reply, marked as answer]
XXL is using CUDA12 libs when regular is using CUDA11. You needed to updated video drivers:

https://www.nvidia.com/download/index.aspx?lang=en-us

**Reply by 1001ruchka** [reply]
> XXL is using CUDA12 libs when regular is using CUDA11.

I have Visual Studio and CUDA 11 installed on my computer. The paths to CUDA 11 libraries are specified in Windows environment variables. Does it make sense to replace CUDA with version 12 if I plan to use XXL version?

**Reply by Purfview** [reply]
I think it doesn't matter what libs you have installed as XXL has needed libs in it's folders.

**Reply by 151SoGood** [reply]
@Purfview Hey Purview, happy new year! I was using Whisper-Faster_r153 for a long time. I wanted to update to the latest version (Faster-Whisper-XXL_r245.1). It wasn't detecting my GPU and used CPU for processing. I note you said that XXL already has the needed libraries, so does that mean we don't need to download CUDA12 libraries from here: https://github.com/Purfview/whisper-standalone-win/releases/tag/libs?

Also, to update to the latest version above, I need to also install the latest CUDA (12.6)? Do you know if we need to do that plus install the nVIDIA CUDA **and** also via pytouch? e.g. "pip3 install torch --index-url https://download.pytorch.org/whl/cu118"

Finally, I noticed the subtitle line length has gotten significantly longer in the latest version, was there some change that happened? Does this mean going forward I would need to specify the length to cap it?

Thanks heaps

**Reply by Purfview** [reply]
Happy new year!

> I note you said that XXL already has the needed libraries, so does that mean we don't need to download CUDA12 libraries from...

For "XXL" you don't need to download those cuBLAS and cuDNN libs, those were for deprecated https://github.com/Purfview/whisper-standalone-win/releases/tag/faster-whisper

>Also, to update to the latest version above, I need to also install the latest CUDA (12.6)?

Update your video driver to support CUDA 12. Probably any CUDA 12.x will be OK.

>Do you know if we need to do that plus install the nVIDIA CUDA and also via pytouch?

You don't need to download anything related to Python. That's written in About - "...for those who don't want to bother with Python."

>Finally, I noticed the subtitle line length has gotten significantly longer in the latest version, was there some change that happened? 

That's just random whisper thing unreleased to any version. You can use `--sentence`, `--standard` ect. to make lines in subs shorter.
